src/sagemaker/apiutils/_base_types.py

```diff
@@ -49,6 +49,9 @@ def from_boto(cls, boto_dict, **kwargs):
             boto_dict (dict): A dictionary of a boto response.
             **kwargs: Arbitrary keyword arguments
         """
+        if not boto_dict:
+            return None
```

Comment:
The original behavior of this public `classmethod` when `boto_dict` was empty was to return an empty `dict`. But your implementation modifies the original behavior and returns `None` instead:

```
>>> boto_dict = {}
>>> if not boto_dict:
...     print("returning None instead of original empty dict")
...
returning None instead of original empty dict
```

This is probably not intended, and would make it harder to deserialize something with no required args in the constructor. You probably want:

```suggestion
        if boto_dict is None:
            return None
```